include/seqan3/range/views/join.hpp

```diff
@@ -7,20 +7,12 @@
 
 /*!\file
  * \author Hannes Hauswedell <hannes.hauswedell AT fu-berlin.de>
- * \brief Provides seqan3::views::join.
+ * \brief [DEPRECATED] Provides seqan3::views::join.
  */
 
 #pragma once
 
-#include <range/v3/view/join.hpp>
+#include <seqan3/utility/views/join_with.hpp>
 
-#include <seqan3/core/platform.hpp>
-#include <seqan3/std/ranges>
-
-namespace seqan3::views
-{
-
-//TODO reimplement me and rename to join_with
-using ::ranges::views::join;
-
-} // namespace seqan3::views
+SEQAN3_DEPRECATED_HEADER(
```

Comment:
```suggestion

namespace seqan3::views
{

/*!\brief A join view.
 * \ingroup views
 * \deprecated Please use std::views::join or seqan3::views::join_with (if a separator is needed)
 */
SEQAN3_DEPRECATED_310 inline constexpr auto join = ::ranges::views::join;

} // namespace seqan3::views

SEQAN3_DEPRECATED_HEADER(
```

Reply:
On second thought, your initial solution might be better in this case.

* as we can deprecate `seqan3/range/views/join.hpp` and
* don't use that internally any more
*  `seqan3/range/views/all.hpp` includes `seqan3/utility/views/join_with.hpp` and thus exposes the deprecated `seqan3::views::join`